Tests/iaas/entropy/entropy-check.py

```diff
@@ -0,0 +1,377 @@
+        wrong = [f"{key}={value}" for key, value in attributes.items() if image.get(key) != value]
+        if wrong:
+            logger.info(f"Image '{image.name}' missing recommended attributes: {', '.join(wrong)}")
+
+
+def check_flavor_attributes(flavors, attributes=FLAVOR_ATTRIBUTES, optional=FLAVOR_OPTIONAL):
+    for flavor in flavors:
+        extra_specs = flavor['extra_specs']
+        wrong = [f"{key}={value}" for key, value in attributes.items() if extra_specs.get(key) != value]
+        miss_opt = [key for key in optional if extra_specs.get(key) is None]
+        if wrong:
+            message = f"Flavor '{flavor.name}' missing recommended attributes: {', '.join(wrong)}"
+            # only report missing optional attributes if recommended are missing as well
+            # reasoning here is that these optional attributes are merely a hint for implementers
+            # and if the recommended attributes are present, we assume that implementers have done their job already
+            if miss_opt:
+                message += f"; additionally, missing optional attributes: {', '.join(miss_opt)}"
+            logger.info(message)
+
+
+def install_test_requirements(fconn):
+    # the following commands seem to be necessary for CentOS 8, but let's not go there
+    # because, frankly, that image is ancient
+    # sudo sed -i -e "s|mirrorlist=|#mirrorlist=|g" /etc/yum.repos.d/CentOS-*
+    # sudo sed -i -e "s|#baseurl=http://mirror.centos.org|baseurl=http://vault.centos.org|g" /etc/yum.repos.d/CentOS-*
+    # Try those commands first that have a high chance of success (Ubuntu seems very common)
+    commands = (
+        # use ; instead of && after update because an error in update is not fatal
+        # also, on newer systems, it seems we need to install rng-tools5...
+        ('aptitude', 'apt-get -v && (sudo apt-get update ; sudo apt-get install -y rng-tools5 || sudo apt-get install -y rng-tools)'),
+        ('dnf', 'sudo dnf install -y rng-tools'),
+        ('yum', 'sudo yum -y install rng-tools'),
+        ('pacman', 'sudo pacman -Syu rng-tools'),
+    )
+    for name, cmd in commands:
+        try:
+            fconn.run(cmd, hide=True)
+        except invoke.exceptions.UnexpectedExit as e:
+            logger.debug(f"Error running '{name}': {e.result.stderr.strip()}")
+        else:
+            return
+    logger.debug("No package manager worked; proceeding anyway as rng-utils might be present nonetheless.")
+
+
+def check_vm_requirements(fconn, image_name):
+    try:
+        entropy_avail = fconn.run('cat /proc/sys/kernel/random/entropy_avail', hide=True).stdout.strip()
+        if entropy_avail != "256":
+            logger.error(
+                f"VM '{image_name}' didn't have a fixed amount of entropy available. "
+                f"Expected 256, got {entropy_avail}."
+            )
+
+        install_test_requirements(fconn)
+        fips_data = fconn.run('cat /dev/random | rngtest -c 1000', hide=True, warn=True).stderr
```

Comment:
I'm confused, didn't we agree on dropping the need for `rngtest`? maybe I'm misremembering?

Reply:
yeah, true; I just remember someone (@fkr ?) saying during the discussion that we should drop the requirement for `rngtest` and I thought there was agreement about this.

maybe I'm misremembering though.

Reply:
I don't think you're necessarily misremembering. We just had multiple discussions, and it's likely that this statement was made (by whomever) in one of them. In the final one, however, we decided to reanimate rngtest because it's an easy way to check whether the CPU instructions are filtered by the hypervisor. I admit that it's not terribly elegant, but it doesn't seem wrong either.